my_sprites.py

```diff
@@ -76,14 +94,38 @@ def go_to_position(self, target_pos: tuple[int, int]):
 
     def on_update(self, delta_time: float = 1 / 60):
 
-        if not self.path:
+        # state control
+        if arcade.has_line_of_sight(self.position, self.target.position, self.barriers.blocking_sprites):
+            self.state = EnemyState.CHASING
+        else:
+            self.state = EnemyState.ROAMING
+
+        # chasing state
+        if self.state == EnemyState.CHASING:
+            self.path = []
+
+            angle_to_target = arcade.get_angle_radians(self.center_x, self.center_y, self.target.center_x, self.target.center_y)
+
+            self.center_x += math.sin(angle_to_target) * self.speed
+            self.center_y += math.cos(angle_to_target) * self.speed
+
+        # roaming state
+        if not self.path and self.state == EnemyState.ROAMING:
```

Comment:
Hmmm.... We should only look at the state... This makes it unclear what the ROAMING state does...

Reply:
so... split up the if statements?

Reply:
Yes, I think so. To make it clear what happens in the _ROAMING_ state.